app/src/main/java/com/nervousfish/nervousfish/modules/constants/Constants.java

```diff
@@ -6,22 +6,29 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.InvalidObjectException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+
 /**
  * Class implementing the main constants for the application.
  */
 public final class Constants implements IConstants {
 
+    private static final long serialVersionUID = -2465684567600871939L;
     private static final Logger LOGGER = LoggerFactory.getLogger("Constants");
     private final static String DB_USERDATA_PATH = "/accountInformation.json";
     private final static String DB_CONTACTS_PATH = "/contacts.json";
-    private final String androidFilesDir;
+    private final IServiceLocator serviceLocator;
+    private transient final String androidFilesDir;
 
     /**
      * Prevents construction from outside the class.
      *
      * @param serviceLocator Can be used to get access to other modules
      */
     private Constants(final IServiceLocator serviceLocator) {
+        this.serviceLocator = serviceLocator;
```

Comment:
Actually, just thought a bit more about why I did this, as consistency is no very good reason. The real reason a bit more serious.
When Constants is serialized we're essentially making a copy of Constants and when we're deserializing it we're essentially turning that copy in a new instance. This can be done using a very unsafe way, using a extralinguistic mechanism, namely the ordinary serialization. However, a much safer way is to do it by using the proxy serialization pattern. This constructs the object the safe way, namely by calling the constructor. The constructor requires an argument unfortunately, namely a service locator. How does the proxy know what this argument should be? Exactly, by storing the value parameter when Contants was created for the first time in a member field!